### PR TITLE
flag to skip arbitrary types of workloads

### PR DESCRIPTION
**Pull request by stevenctl**
Alternative/addition to #35233

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/util/sets/string.go [hunk 1]
```diff
@@ -139,3 +146,15 @@ func (s Set) Equals(other Set) bool {
 func (s Set) Empty() bool {
 	return len(s) == 0
 }
+
+func (s Set) Set(value string) error {
+	parts := strings.Split(value, ",")
+	for _, part := range parts {
```
Comment (hzxuzhonghu) [inline pilot/pkg/util/sets/string.go:152]: What's this, i haven't looked at kind of set like this

Comment (hzxuzhonghu) [reply pilot/pkg/util/sets/string.go:152]: It will lead to misuse.

Comment (howardjohn) [reply pilot/pkg/util/sets/string.go:152]: Oh yeah I didn't notice this was in the core code. This should probably just be done in the test code to make sure no one uses this in core code?

Comment (hzxuzhonghu) [reply pilot/pkg/util/sets/string.go:152]: Agree

Comment (stevenctl) [reply pilot/pkg/util/sets/string.go:152]: Like `type setFlag sets.Set` or something +1

**Pull request merged**